_posts/2022-12-08-developer-survey-2022.md

```diff
@@ -0,0 +1,123 @@
+
+## Platforms and Deployment targets
+
+When our cloud projects get live, we deploy them most often in Microsoft Azure and Amazon AWS (roughly 33% for each) and Google GCP (12%). Some 10% are deployed in private cloud setups and 10% in Windows Server. A number of developers are also indicating interest towards deploying in IBM Cloud, which is currently not used in the projects.
+
+For deployment setup and configuration, we use Docker (65%), Kubernetes (15%), VirtualBox (8%), Podman (5%) and Vagrant (5%).
+
+During past two years, Solita developers have done desktop applications for Windows, Linux and MacOS, even though the volumes are much lower than in more common browser based solutions. 
+
+In the mobile context, we have done iOS/Android apps, some wearable OS and assistant solutions, plus a handful of Raspberry Pi / Arduino projects. There is also a good amount of interest among the personnel to get more work in the mobile/embedded projects.
+
+## DevOps
+
+For version control, git has swept the floor - in the survey we did not have a single vote for any other version control system. In the VCS service selection, there was more variation. GitHub was the most used (Solita is a GH corporate customer). Where the customer has other preferences, we have used Azure DevOps Services, GitLab and BitBucket.
+
+In the devops section we had an open question about the use of git hooks in the daily work.
+
+The build and deployment pipelines are most often built with either Azure Pipelines (35%), Jenkins (28%), or GitHub Actions (20%). Some projects use GitLab CI (7%), or do deployments with custom scripting (5%). The rest consists of 4-5 votes per platform in the list of CircleCI, BitRise, Octopus Deploy and TeamCity.
+
+For infrastructure management, the top votes spread over various shell/makefile/yaml/powershell solutions, Terraform, Ansible and AWS CDK. Some projects make use of other Azure products like CloudFormation and Resource manager or Google Cloud Deployment Manager.
+
+Majority of the projects have automated deployment of the software to test environments. For production environments, it is more common to have automated deployment with a manual trigger, but 25% of respondents indicate that in their projects also the production deployment is fully automatic.
+
+## Testing & Monitoring
+
+While in the title catalog in the Solita developer community we do not have many people dedicated testing roles, we do have proper focus in the maturity and quality of the code that we produce. Our teams contribute in all testing activities from smoke testing to regression/acceptance tests and we also conduct penetration testing and A/B testing for some of our customers.
+
+In the past two years the most used test frameworks are Cypress (31%), JUnit (30%) and Jest (21%). Besides unit testing, UI testing and E2E testing are done with e.g. Robot Framework, Puppeteer and Cucumber frameworks.
```

Comment:
I think this could have better wording, as it might be read as Cypress would be used for unit testing. 

I would perhaps mention Cypress again in the E2E section, as well as Playwright (I think which got bunch of mentions).

Additionally, I would expand upon "E2E" to `end-to-end (E2E)` for the first time it is used.

Reply:
Fixed